api/tasks.py

```diff
@@ -41,10 +42,13 @@ async def poll_updates(obj: Union[int, models.Invoice], task_wallets: Dict[str,
     await process_invoice(obj, task_wallets, payment_methods)
 
 
-async def process_invoice(invoice: models.Invoice, task_wallets: Dict[str, str], payment_methods: List[models.PaymentMethod]):
+async def process_invoice(
+    invoice: models.Invoice, task_wallets: Dict[str, str], payment_methods: List[models.PaymentMethod], notify: bool = True
+):
     while not settings.shutdown.is_set():
         for method in payment_methods:
-            invoice_data = await method.coin.getrequest(method.payment_address)
+            invoice_data = method.coin.getrequest(method.payment_address)
+            invoice_data = await invoice_data if inspect.isawaitable(invoice_data) else invoice_data
```

Comment:
I think that tests should be updated to return a coroutine, instead of modifying tasks